### PR TITLE
#165020191 Configure PostgreSQL as database engine

### PR DESCRIPTION
**Pull request by avpaul**
## What does this PR do?
Configure PostgreSQL as the database engine

## Description of Task to be completed?
Authors Haven comes with support for MongoDB, a NoSQL Database. We are changing to PostgreSQL,  an Open Source fully featured and optimized SQL database engine. To make the process of working with an SQL database easier we setup Sequelize as an ORM.

## How should this be manually tested?
After cloning the repo
```
    git checkout -t ch-165020191-postgresql-migration
    npm i
    // add a .env file with DATABASE_URL variable 
    npm start
```

## What are the relevant pivotal tracker stories?
[#165020191 Configure Authors Haven to use PostgreSQL as DB Engine](https://www.pivotaltracker.com/n/projects/2321840)

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "use_env_variable": "DATABASE_URL",
+    "dialect": "postgres",
+    "logging": false
+  },
+  "test": {
+    "username": "root",
+    "password": null,
+    "database": "database_test",
+    "host": "127.0.0.1",
+    "dialect": "mysql"
```
Comment (shemaeric) [inline config/config.json]: @avpaul I think during the test, we will be using the same database which is Postgres rather than Mysql.

File: .sequelizerc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require('@babel/register');
```
Comment (aaronsekisambu) [inline .sequelizerc]: @avpaul shouldn't this conform to ES6 standards?

File: .sequelizerc [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+const path = require('path');
+
+module.exports = {
```
Comment (aaronsekisambu) [inline .sequelizerc]: @avpaul shouldn't this conform to ES6 standards?

File: .sequelizerc (hunk 2 above)
Comment (avpaul) [reply .sequelizerc]: Thanks! consider it done

File: models/user.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import crypto from 'crypto';
+
+module.exports = (sequelize, DataTypes) => {
+  const User = sequelize.define('User', {
```
Comment (shemaeric) [inline models/user.js:4]: for good naming, It would be better if the table name(User) was in small letters.

File: models/user.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import crypto from 'crypto';
+
+module.exports = (sequelize, DataTypes) => {
```
Comment (shemaeric) [inline models/user.js]: Because you have decided to go with es6 syntax, it would be better not to use module.exports but export synthax of es6

**Review by daud1 — CHANGES_REQUESTED** [review]
Please implement the changes requested by HoundCI and Aaron before I can review.

File: config/passport.js [hunk 6]
```diff
@@ -1,26 +1,3 @@
 const passport = require("passport");
 const LocalStrategy = require("passport-local").Strategy;
-const mongoose = require("mongoose");
-const User = mongoose.model("User");
 
-passport.use(
```
Comment (daud1) [inline config/passport.js]: please revert this change.
I don't think all this code should be removed.

**Review by daud1 — CHANGES_REQUESTED** [review]

File: config/passport.js (hunk 6 above)
Comment (daud1) [reply config/passport.js]: Also, fix all merge conflicts by rebasing to develop.

Comment (avpaul) [reply config/passport.js]: The changes have been reverted

Comment (avpaul) [reply config/passport.js]: The changes have been reverted and I've rebased

**Review by niyobobo — DISMISSED** [review]
Apart from those HoundCI errors, other LGTM.

**Review by fridolinho — DISMISSED** [review]

File: models/user.js (hunk 4 above)
Comment (avpaul) [reply models/user.js:4]: I think as we are giving the name of the exported model and models are like classes the name should be in a capital letter!

**Review by shemaeric — APPROVED** [review]
All the requested changes are applied and it looks good.

**Review by niyobobo — APPROVED** [review]

**Review by aaronsekisambu — APPROVED** [review]

**Review by daud1 — APPROVED** [review]
